### PR TITLE
 Making unit tests for BattleHex and some estetic improvements.

### PR DESCRIPTION
**Pull request by FeniksFire**
Making unit tests for BattleHex.
Delete unnecessary range, templates.

### CODE REVIEW

File: test/CVcmiTestConfig.cpp [hunk 1]
```diff
@@ -33,10 +33,10 @@ CVcmiTestConfig::CVcmiTestConfig()
 	loadDLLClasses();
 	logGlobal->info("Initialized global test setup.");
 
-	const std::string TEST_DATA_DIR = "test/";
-
-	auto loader = new CFilesystemLoader("test/", TEST_DATA_DIR);
-	dynamic_cast<CFilesystemList*>(CResourceHandler::get())->addLoader(loader, false);
+// These lines make a "Test setup error:"
+//	const std::string TEST_DATA_DIR = "test/";
```
Comment (alexvins) [inline test/CVcmiTestConfig.cpp]: May be this path is incorrect in your build.

Comment (FeniksFire) [inline test/CVcmiTestConfig.cpp]: If so, more people have that same problem, and this is wrong so should be comment out, I can or someone else can uncommend it when look in closer for it  and fix it ;)

Comment (alexvins) [inline test/CVcmiTestConfig.cpp]: Where is tests executable in your build?

Comment (alexvins) [inline test/CVcmiTestConfig.cpp]: Try const std::string TEST_DATA_DIR = "testdata/";

Comment (FeniksFire) [inline test/CVcmiTestConfig.cpp]: Not working. Here you have somes screens:
http://imgur.com/a/KmAqp

**Comment by DjWarmonger** [comment]
@FeniksFire Oh I didn't know that. If they are supported by standard now, we can definitelly drop boost here.

File: test/CVcmiTestConfig.cpp (hunk 1 above)
Comment (alexvins) [inline test/CVcmiTestConfig.cpp]: Try `const std::string TEST_DATA_DIR = ".";`

In your build "terrainViewMappings.json" & "TerrainViewTest.h3m" are in same directory with vcmitest.

Comment (FeniksFire) [inline test/CVcmiTestConfig.cpp]: Huh, It's works. Thanks for tips. These differences seem to stem from differences in project file, but I'm not sure. I will therefore have to do something about it.

Comment (alexvins) [inline test/CVcmiTestConfig.cpp]: Sorry, I`ve done this on purpose about a year ago. The idea was to make tests work built in-place, I`ll do smth more smart later.

Comment (alexvins) [inline test/CVcmiTestConfig.cpp]: Its better to enclose `auto loader = new CFilesystemLoader("test/", TEST_DATA_DIR);` with ifdef for now.

Comment (FeniksFire) [inline test/CVcmiTestConfig.cpp]: well ok, I enclose it and I give short info about these lines, about the path :)

Comment (FeniksFire) [inline test/CVcmiTestConfig.cpp]: I do it without ifdef although a similar effect, simple check if directory exist.

**Comment by FeniksFire** [comment]
there is anything I should correct in this pull? why don't merge it?

**Comment by DjWarmonger** [comment]
@alexvins , it's up to you.

**Pull request merged**